scripts/safe_template_linter.py

```diff
@@ -398,23 +399,42 @@ def _check_mako_file_is_safe(self, mako_template, results):
             results: A file results objects to which violations will be added.
 
         """
-        has_page_default = self._has_page_default(mako_template, results)
-        if not has_page_default:
-            results.violations.append(RuleViolation(Rules.mako_missing_default))
+        has_page_default = False
+        if self._has_multiple_page_tags(mako_template):
+            results.violations.append(RuleViolation(Rules.mako_multiple_page_tags))
+        else:
+            has_page_default = self._has_page_default(mako_template)
+            if not has_page_default:
+                results.violations.append(RuleViolation(Rules.mako_missing_default))
         self._check_mako_expressions(mako_template, has_page_default, results)
         results.prepare_results(mako_template)
 
-    def _has_page_default(self, mako_template, results):
+    def _has_multiple_page_tags(self, mako_template):
+        """
+        Checks if the Mako template contains more than one page expression.
+
+        Arguments:
+            mako_template: The contents of the Mako template.
+
+        """
+        matches = 0
+        for match in re.finditer('<%page ', mako_template, re.IGNORECASE):  # pylint: disable=unused-variable
+            matches += 1
+            if matches == 2:
+                return True
+        return False
+
+    def _has_page_default(self, mako_template):
         """
         Checks if the Mako template contains the page expression marking it as
         safe by default.
 
         Arguments:
             mako_template: The contents of the Mako template.
-            results: A list of results into which violations will be added.
 
         """
-        page_h_filter_regex = re.compile('<%page expression_filter=(?:"h"|\'h\')\s*/>')
+        # could match across tags, but this should be good enough.
+        page_h_filter_regex = re.compile('<%page.*expression_filter=(?:"h"|\'h\').*/>')
```

Comment:
To not match across tags (if you care), try
`re.compile('<%page[^<]*expression_filter=(?:"h"|\'h\').*?/>')`